### PR TITLE
Add support for Magnetometer and logging on Anyleaf H7

### PR DESCRIPTION
**Pull request by David-OConnor**

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/hwdef/AnyleafH7/defaults.parm [hunk 1]
```diff
@@ -20,4 +20,7 @@ SERIAL3_PROTOCOL 16 # ESC telemetry
 
 
 # Remove default serial protocols; set to None.
-SERIAL6_PROTOCOL -1
\ No newline at end of file
+SERIAL6_PROTOCOL -1
+
+# Set the logging backend to flash memory
+LOG_BACKEND_TYPE 4
```
Comment (andyp1per) [inline libraries/AP_HAL_ChibiOS/hwdef/AnyleafH7/defaults.parm:26]: The way to do this is:
```
define HAL_LOGGING_DATAFLASH_ENABLED 1
```
in the hwdef

**Review by andyp1per — CHANGES_REQUESTED** [review]
Need to rebase and clean up the commit messages

**Comment by David-OConnor** [comment]
Here goes nothing...

**Comment by David-OConnor** [comment]
I have tried:

-`git rebase -i HEAD~6`
- In the text editor, replace all `pick ` with `s ` except for the one at the top.

I am getting errors about detached head, not being on a branch, conflicts with arducopter.py  etc. 

How would you handle this?

File: libraries/AP_HAL_ChibiOS/hwdef/AnyleafH7/defaults.parm (hunk 1 above)
Comment (David-OConnor) [reply libraries/AP_HAL_ChibiOS/hwdef/AnyleafH7/defaults.parm]: Great; thanks! Fixed.

**Comment by David-OConnor** [comment]
Hmm. This is what I'm seeing now:

```
PS C:\Users\the_a\Desktop\t3\ardupilot> git pull --rebase upstream master

fatal: 'upstream' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.


PS C:\Users\the_a\Desktop\t3\ardupilot> git pull --rebase origin master
From https://github.com/david-oconnor/ardupilot

 * branch                  master     -> FETCH_HEAD
Auto-merging Tools/autotest/arducopter.py
CONFLICT (content): Merge conflict in Tools/autotest/arducopter.py
error: could not apply 1bfd640923... autotest: add test for Copter compassmot
hint: Resolve all conflicts manually, mark them as resolved with
hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
hint: You can instead skip this commit: run "git rebase --skip".
hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
Could not apply 1bfd640923... autotest: add test for Copter compassmot
PS C:\Users\the_a\Desktop\t3\ardupilot>

```

**Comment by andyp1per** [comment]
> Hmm. This is what I'm seeing now:
> 
> ```
> PS C:\Users\the_a\Desktop\t3\ardupilot> git pull --rebase upstream master
> 
> fatal: 'upstream' does not appear to be a git repository
> fatal: Could not read from remote repository.
> 
> Please make sure you have the correct access rights
> and the repository exists.
> 
> 
> PS C:\Users\the_a\Desktop\t3\ardupilot> git pull --rebase origin master
> From https://github.com/david-oconnor/ardupilot
> 
>  * branch                  master     -> FETCH_HEAD
> Auto-merging Tools/autotest/arducopter.py
> CONFLICT (content): Merge conflict in Tools/autotest/arducopter.py
> error: could not apply 1bfd640923... autotest: add test for Copter compassmot
> hint: Resolve all conflicts manually, mark them as resolved with
> hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
> hint: You can instead skip this commit: run "git rebase --skip".
> hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
> Could not apply 1bfd640923... autotest: add test for Copter compassmot
> PS C:\Users\the_a\Desktop\t3\ardupilot>
> ```

git remote add upstream https://github.com/ArduPilot/ardupilot.git

**Comment by David-OConnor** [comment]
Thank you.

```
PS C:\Users\the_a\Desktop\t4\ardupilot> git pull --rebase upstream master

From https://github.com/ArduPilot/ardupilot
 * branch                  master     -> FETCH_HEAD
 * [new branch]            master     -> upstream/master
Auto-merging ArduCopter/mode_guided.cpp
Auto-merging ArduPlane/Parameters.cpp
Auto-merging ArduPlane/quadplane.cpp
Auto-merging ArduPlane/system.cpp
CONFLICT (content): Merge conflict in ArduPlane/system.cpp
Auto-merging Rover/Parameters.cpp
Auto-merging Tools/AP_Bootloader/board_types.txt
CONFLICT (content): Merge conflict in Tools/AP_Bootloader/board_types.txt
Auto-merging Tools/autotest/arducopter.py
CONFLICT (content): Merge conflict in Tools/autotest/arducopter.py
Auto-merging Tools/autotest/arduplane.py
Auto-merging Tools/autotest/rover.py
CONFLICT (content): Merge conflict in Tools/autotest/rover.py
Auto-merging Tools/autotest/vehicle_test_suite.py
Auto-merging libraries/AP_NavEKF3/AP_NavEKF3_Control.cpp
Auto-merging libraries/AP_NavEKF3/AP_NavEKF3_core.h
Auto-merging libraries/AP_Param/AP_Param.cpp
Auto-merging libraries/AP_Scripting/applets/revert_param.lua
Auto-merging libraries/AP_Scripting/docs/docs.lua
CONFLICT (content): Merge conflict in libraries/AP_Scripting/docs/docs.lua
Auto-merging libraries/AP_Scripting/examples/fault_handling.lua
CONFLICT (add/add): Merge conflict in libraries/AP_Scripting/examples/fault_handling.lua
Auto-merging libraries/AP_Scripting/generator/src/main.c
CONFLICT (content): Merge conflict in libraries/AP_Scripting/generator/src/main.c
Auto-merging libraries/SITL/SITL.cpp
CONFLICT (content): Merge conflict in libraries/SITL/SITL.cpp
Auto-merging libraries/SITL/SITL.h
error: could not apply 1bfd640923... autotest: add test for Copter compassmot
hint: Resolve all conflicts manually, mark them as resolved with
hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
hint: You can instead skip this commit: run "git rebase --skip".
hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
Could not apply 1bfd640923... autotest: add test for Copter compassmot
PS C:\Users\the_a\Desktop\t4\ardupilot>
```

**Comment by andyp1per** [comment]
at this point I suggest you create a new fresh branch, cherry-pick in your two commits and then rebase. The merge commits have really messed you up.

**Comment by peterbarker** [comment]
We merged the alternative PR which @David-OConnor created.

Closing this one.

**Pull request closed without merging**